packages/client/hmi-client/src/services/notificationEventHandlers.ts

```diff
@@ -3,19 +3,36 @@ import {
 	ClientEventType,
 	ExtractionStatusUpdate,
 	ProgressState,
-	TaskResponse
+	TaskResponse,
+	TaskStatus
 } from '@/types/Types';
 import { logger } from '@/utils/logger';
 import { Ref } from 'vue';
-import { NotificationItem } from '@/types/common';
+import { NotificationItem, NotificationItemStatus } from '@/types/common';
+import { snakeToCapitalSentence } from '@/utils/text';
 import { getDocumentAsset } from './document-assets';
 
-export const getStatus = (data: { error: string; t: number }) => {
+export const getStatusFromProgress = (data: { error: string; t: number }) => {
 	if (data.error) return ProgressState.Failed;
 	if (data.t >= 1.0) return ProgressState.Complete;
 	return ProgressState.Running;
 };
 
+const isTaskResponse = (data: any): data is TaskResponse =>
+	data.id !== undefined && data.script !== undefined && data.status;
+
+const getStatusFromTaskResponse = (data: TaskResponse) => {
+	const statusMap = {
+		[TaskStatus.Queued]: ProgressState.Queued,
+		[TaskStatus.Running]: ProgressState.Running,
+		[TaskStatus.Success]: ProgressState.Complete,
+		[TaskStatus.Failed]: ProgressState.Failed,
+		[TaskStatus.Cancelled]: ProgressState.Cancelled,
+		[TaskStatus.Cancelling]: ProgressState.Cancelling
+	};
```

Comment:
@dvince2 @jryu01 So does this mean that `ProgressState` is the canonical state we are going with? Maybe a little comment here would help?

Also it seems unnecessary to (re)create this lookup every time the function is called, can we move it outside and treat statusMap as static?

Reply:
Done